lib/connect.js

```diff
@@ -234,25 +240,86 @@ const getTasks = (model, setupListeners) => {
         setupListeners(mongoClient);
       }
 
+      let maybeTunnelError;
+
+      if (tunnel) {
+        tunnel.on('error', (tunnelError) => {
+          if (tunnelError.origin === 'ssh-client') {
+            maybeTunnelError = tunnelError;
+          }
+        });
+      }
+
       mongoClient.connect((err, _client) => {
```

Comment:
now that the tunnel uses promises, what about having an `async function connect() {}` that we export with `util.callbackify` instead than manually converting all of the promise call manually?

Reply:
This would not be a 1 to 1 replacement, `connect` method synchronously returns `state` emitter and additionally allows to pass a callback that will evaluate when connection happened, can't really replace this behavior with promises